receiver/mysqlreceiver/scraper.go

```diff
@@ -94,12 +94,12 @@ func (m *mySQLScraper) scrape(context.Context) (pmetric.Metrics, error) {
 	globalStats, err := m.sqlclient.getGlobalStats()
 	if err != nil {
 		m.logger.Error("Failed to fetch global stats", zap.Error(err))
-		return pmetric.Metrics{}, err
+		errs.AddPartial(1, err)
+	} else {
+		m.recordDataPages(now, globalStats, errs)
+		m.recordDataUsage(now, globalStats, errs)
 	}
```

Comment:
I decided to make a change in handling errors here, too.
The `else` branch is necessary, because `recordDataPages` and `recordDataUsage` don't check for null.

Reply:
Update: actually, for a map `m`, `m["a"]` will return empty string, so these functions would report 2 errors each. I'm not sure what is the best solution here, what do you think?